Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -72,98 +136,188 @@ struct RedundantTypeAnnotationRule: OptInRule, SubstitutionCorrectableRule {
+                  let typeAnnotation = node.typeAnnotation,
+                  let initializer = node.initializer,
+                  typeAnnotation.isRedundant(for: configuration, initializerExpr: initializer.value)
+            else {
+                return super.visit(node)
             }
-            .filter { !isFalsePositive(file: file, range: $0.0) }
-            .filter { !isIBInspectable(file: file, range: $0.0) }
-            .compactMap {
-                file.match(pattern: typeAnnotationPattern,
-                           excludingSyntaxKinds: SyntaxKind.commentAndStringKinds, range: $0.0).first
+
+            correctionPositions.append(typeAnnotation.positionAfterSkippingLeadingTrivia)
+
+            // Add a leading whitespace to the initializer sequence so there is one
+            // between the variable name and the '=' sign
+            let initializerWithLeadingWhitespace = initializer
+                .with(\.leadingTrivia, Trivia.space)
+
+            return super.visit(
+                node.with(\.typeAnnotation, nil)
+                    .with(\.initializer, initializerWithLeadingWhitespace)
+            )
+        }
+
+        override func visit(_ node: OptionalBindingConditionSyntax) -> OptionalBindingConditionSyntax {
+            guard let typeAnnotation = node.typeAnnotation,
+                  let initializer = node.initializer,
+                  typeAnnotation.isRedundant(for: configuration, initializerExpr: initializer.value)
+            else {
+                return super.visit(node)
             }
-    }
 
-    private func isFalsePositive(file: SwiftLintFile, range: NSRange) -> Bool {
-        guard let typeNames = getPartsOfExpression(in: file, range: range) else { return false }
+            correctionPositions.append(typeAnnotation.positionAfterSkippingLeadingTrivia)
 
-        let lhs = typeNames.variableTypeName
-        let rhs = typeNames.assigneeName
+            // Add a leading whitespace to the initializer sequence so there is one
+            // between the variable name and the '=' sign
+            let initializerWithLeadingWhitespace = initializer
+                .with(\.leadingTrivia, Trivia.space)
 
-        if lhs == rhs || (lhs == "Bool" && (rhs == "true" || rhs == "false")) {
-            return false
-        } else {
-            return true
+            return super.visit(
+                node.with(\.typeAnnotation, nil)
+                    .with(\.initializer, initializerWithLeadingWhitespace)
+            )
         }
     }
+}
 
-    private func getPartsOfExpression(
-        in file: SwiftLintFile, range: NSRange
-    ) -> (variableTypeName: String, assigneeName: String)? {
-        let substring = file.stringView.substring(with: range)
-        let components = substring.components(separatedBy: "=")
+private extension TypeAnnotationSyntax {
+    func isRedundant(for configuration: RedundantTypeAnnotationConfiguration,
```

Comment:
`configuration` not needed in here, is it?

Reply:
Indeed!